packages/react-components/react-card/src/components/Card/useCardSelectable.ts

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import { Enter, Space } from '@fluentui/keyboard-keys';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import { useFocusFinders } from '@fluentui/react-tabster';
+
+import type { CardOnSelectEvent, CardProps, CardRefElement } from './Card.types';
+
+export const useCardSelectable = (props: CardProps, cardRef: React.RefObject<CardRefElement>) => {
+  const { select, selected, defaultSelected, onCardSelect } = props;
+
+  const { findAllFocusable } = useFocusFinders();
+  const selectableRef = React.useRef<HTMLDivElement>(null);
+
+  const isSelectable = [selected, defaultSelected, onCardSelect, select].some(bool => typeof bool !== 'undefined');
+  const hasSelectSlot = Boolean(select);
+
+  const [isCardSelected, setIsCardSelected] = React.useState(false);
+
+  const shouldRestrictTriggerAction = React.useCallback(
+    (event: CardOnSelectEvent) => {
+      if (cardRef.current && selectableRef.current) {
```

Comment:
is all the logic in this check really necessary ? you should just be able to do `e.currentTarget === e.target` to make sure that the event was triggered on the card itself and not its children